queryset_sequence/__init__.py

```diff
@@ -460,6 +460,11 @@ def __init__(self, *args):
         self._iterable_class = ModelIterable
         self._result_cache = None
 
+        self._model = kwargs.get("model", None)
+        # add missing DoesNotExist to abstract models
+        if self._model and self._model._meta.abstract:
+            self._model.DoesNotExist = ObjectDoesNotExist
```

Comment:
This will modify the actual object passed in, right? That seems not awesome.

(E.g. `MyAbstractModel` will now have a `DoesNotExist` set on it.)

It might make more sense to store the exception class itself on `QuerySetSequence`, e.g. calculate `_does_not_exist` once here and then use it.

Reply:
Yes, this does modify the model. I also think that this is far from ideal but DRF does only handle `model.DoesNotExist` exceptions (see https://github.com/django/django/blob/f9ec777a826816e20e68021c0e73b5a76be650af/django/shortcuts.py#L86). For normal models this will be fine. But abstract models do not have this exception and won't work (which is fixed by the code).

Therefore I don't see another option besides modifying the model here but I'm open to suggestions.

Reply:
The only other thing I can think of is using some sort of proxy for the model:

```python
class ProxyModel:
    def __init__(self, model=None):
        self._model = model

        if self._model:
            self.DoesNotExist = model.DoesNotExist
        else:
            self.DoesNotExist = ObjectDoesNotExist

	def __getattr__(self, name):
        return getattr(self._model, name)
```

And then in the `QuerySetSequence` we would do `self.model = ProxyModel(model)`.

This can get finicky, but at least gives a bit more control and doesn't monkey-patch anything.

Reply:
This is a pretty good idea in my opinion. I will try this out and report back!